server/models/groups/createGroup.js

```diff
@@ -0,0 +1,53 @@
+const { pool } = require('../database')
+
+/**
+ *z
+ * @param {*} param0
+ * @param {String} param0.username
+ * @param {Object} param0.body
+ * @return {Promise}
+ *
+ */
+
+function createGroup({ username, body }) {
+  return new Promise((resolve, reject) => {
+    const { group_name: groupName, confidential, members } = body
+    pool.query(
+      `INSERT INTO cl_judge.groups (group_name,creator,confidential,member_count)
+        VALUES(?,?,?,?)`,
+      [groupName, username, confidential, members.length + 1],
+      (error, results) => {
+        if (error) {
+          return reject(error)
+        }
+        const { insertId } = results
+        pool.query(
+          `INSERT INTO cl_judge.user_groups (username,group_id,is_group_moderator) VALUES(?,?,?)`,
+          [username, insertId, 1],
+          (error) => {
+            if (error) {
+              return reject(error)
+            }
+          }
+        )
+        members.forEach((member) => {
+          pool.query(
+            `INSERT INTO cl_judge.user_groups (username,group_id,is_group_moderator) VALUES(?,?,?)`,
+            [member, insertId, 0],
+            (error) => {
+              if (error) {
+                return reject(error)
```

Comment:
This should not reject the entire process of creating group, due to just 1 invalid username.
We can do something like maintain a variable `count` and decrement it on an unsuccessful member request.

Reply:
And then update `member_count` in `groups` table